### PR TITLE
Add a `splunk` sink operator

### PR DESCRIPTION
**Pull request by IyeOnline**
This PR implements a splunk sink operator, which sends to a Splunk HEC.

- Fixes https://github.com/tenzir/issues/issues/1899

### CODE REVIEW

File: web/docs/tql2/operators/save_splunk.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
+# save_splunk
+
+Sends events to a [Splunk HEC](https://docs.splunk.com/Documentation/Splunk/9.3.1/Data/UsetheHTTPEventCollector)
+
+```tql
+save_splunk url:str, hec_token=str,
```
Comment (jachris) [inline web/docs/tql2/operators/save_splunk.md:6]: This should probably be `to_splunk`, as the `save_*` operators only take bytes.

File: web/docs/tql2/operators/splunk.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+### `sourcetype = expr (optional)`
+
+An optional expression for [Splunk's
+`sourcetype`](https://docs.splunk.com/Splexicon:Sourcetype) that evaluates to a
+`string`. You can use this to set the `sourcetype` per event.
+
+Regardless of the chosen `sourcetype`, the event itself is given as the value to
+the `event` key in the top level object being send.
+
+Defaults to `_json`.
```
Comment (raxyte) [inline web/docs/tql2/operators/splunk.md:54]: Maybe link to https://docs.splunk.com/Documentation/Splunk/9.3.1/Data/Listofpretrainedsourcetypes as list of possible sourcetypes.

**Review by raxyte — APPROVED** [review]

File: web/docs/tql2/operators/splunk.md (hunk 2 above)
Comment (IyeOnline) [reply web/docs/tql2/operators/splunk.md:54]: I just thought about this some more, and I now think that we maybe *shouldnt* do that, as we dont actually support those source types as-is. We always write the entire body into the top-level `event` key, which isnt correct for more of the predefined ones.

File: changelog/next/features/4719--splunk-sink.md [hunk 3]
```diff
@@ -0,0 +1 @@
+The new `splunk` sink operator writes data to Splunk HEC endpoint.
```
Comment (jachris) [inline changelog/next/features/4719--splunk-sink.md:1]: This needs to be change to the actual name of the operator.

**Pull request merged**